drivers/timer/hpet.c

```diff
@@ -208,9 +218,14 @@ static inline void hpet_timer_conf_set(uint32_t val)
  *
  * @param val Value to be written to the register
  */
-static inline void hpet_timer_comparator_set(uint32_t val)
+static inline void hpet_timer_comparator_set(uint64_t val)
 {
-	sys_write32(val, TIMER0_COMPARATOR_REG);
+#if CONFIG_X86_64
```

Comment:
using if (IS_ENABLED(CONFIG_X86_64)) ... would be nice here.

Reply:
only X86_64 has sys_write64() function. #if MUST bs used, or there will be compiling error.

Reply:
indeed